### PR TITLE
Re-strucutre persist tool and remove redundant modules

### PR DESCRIPTION
**Pull request by MadhukaHarith92**

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -2,66 +2,57 @@ name: Publish release
...
           GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
           BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_ACCESS_TOKEN }}
           packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
           packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
-          publishUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
-          publishPAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
         run: |
-          ./gradlew clean release -Prelease.useAutomaticVersion=true
+          ./gradlew release -Prelease.useAutomaticVersion=true
           ./gradlew -Pversion=${VERSION} publish -x test -PpublishToCentral=true
```
Comment (niveathika) [inline .github/workflows/publish-release.yml:42]: Unnecessary flag

**Review by niveathika — COMMENTED** [review]
Lets clean up gitignore file too. I saw some files in target folder being committed

File: persist-cli/build.gradle [hunk 2]
```diff
@@ -20,7 +20,7 @@ plugins {
     id 'checkstyle'
 }
 
-description = 'Ballerina - Persist-Tools Java Native'
+description = 'Ballerina - Persist-CLI Java Native'
```
Comment (niveathika) [inline persist-cli/build.gradle:23]: Review dependencies for cli and cli-test modules. IMO some of them are unnecessary

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -2,66 +2,57 @@ name: Publish release
 
 on:
   workflow_dispatch:
-  repository_dispatch:
-    types: [ stdlib-release-pipeline ]
 
 jobs:
   publish-release:
     runs-on: ubuntu-latest
-    if: github.repository_owner == 'ballerina-platform'
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply .github/workflows/publish-release.yml]: Fixed in [679ed08](https://github.com/ballerina-platform/persist-tools/pull/30/commits/679ed08c603c0cff63a00ad704fef60fd3e62dc2)

File: persist-cli-tests/build.gradle [hunk 4]
```diff
@@ -26,26 +26,80 @@ description = 'Ballerina - persist-tools Tooling Test'
...
+
 dependencies {
     checkstyle project(':checkstyle')
     checkstyle "com.puppycrawl.tools:checkstyle:${checkstylePluginVersion}"
 
-    testImplementation group: 'org.ballerinalang', name: 'ballerina-runtime', version: "${ballerinaLangVersion}"
-    testImplementation group: 'org.ballerinalang', name: 'ballerina-tools-api', version: "${ballerinaLangVersion}"
-    testImplementation (group: 'org.ballerinalang', name: 'ballerina-cli', version: "${ballerinaLangVersion}") {
-        exclude group: 'org.ballerinalang', module: 'maven-resolver'
+    implementation "org.ballerinalang:ballerina-lang:${ballerinaLangVersion}"
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply persist-cli-tests/build.gradle]: Updated in [cc080c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/cc080c2cd0a96d359e3d950202a76cb2637f7405)

File: persist-cli-tests/build.gradle [hunk 5]
```diff
@@ -26,26 +26,80 @@ description = 'Ballerina - persist-tools Tooling Test'
...
 }
 
+def bDistribution = file("$project.buildDir/extracted-distribution/jballerina-tools-${ballerinaLangVersion}")
+
+task jBallerinaPack {
+    doLast {
+        configurations.balTools.resolvedConfiguration.resolvedArtifacts.each { artifact ->
+            copy {
+                from project.zipTree(artifact.getFile())
+                into new File(project.buildDir, "extracted-distribution/")
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply persist-cli-tests/build.gradle]: Updated in [cc080c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/cc080c2cd0a96d359e3d950202a76cb2637f7405)

File: persist-cli-tests/build.gradle [hunk 6]
```diff
@@ -26,26 +26,80 @@ description = 'Ballerina - persist-tools Tooling Test'
...
+    outputs.dir bDistribution
+}
+
+task unpackStdLibs() {
+    dependsOn(jBallerinaPack)
+    doLast {
+        configurations.ballerinaStdLibs.resolvedConfiguration.resolvedArtifacts.each { artifact ->
+            copy {
+                from project.zipTree(artifact.getFile())
+                into new File("${project.buildDir}/extracted-distribution/" + artifact.name + "-zip")
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply persist-cli-tests/build.gradle]: Updated in [cc080c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/cc080c2cd0a96d359e3d950202a76cb2637f7405)

File: persist-cli-tests/build.gradle [hunk 7]
```diff
@@ -26,26 +26,80 @@ description = 'Ballerina - persist-tools Tooling Test'
...
+            }
+        }
+    }
+}
+
+task copyStdlibs(type: Copy) {
+    dependsOn(unpackStdLibs)
+    def ballerinaDist = "$project.buildDir/extracted-distribution/jballerina-tools-${ballerinaLangVersion}"
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply persist-cli-tests/build.gradle]: Updated in [cc080c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/cc080c2cd0a96d359e3d950202a76cb2637f7405)

File: persist-cli-tests/build.gradle [hunk 8]
```diff
@@ -26,26 +26,80 @@ description = 'Ballerina - persist-tools Tooling Test'
...
+    into ballerinaDist
+
+    /* Standard Libraries */
+    configurations.ballerinaStdLibs.resolvedConfiguration.resolvedArtifacts.each { artifact ->
+        def artifactExtractedPath = "${project.buildDir}/extracted-distribution/" + artifact.name + "-zip"
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply persist-cli-tests/build.gradle]: Updated in [cc080c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/cc080c2cd0a96d359e3d950202a76cb2637f7405)

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (MadhukaHarith92) [reply .github/workflows/publish-release.yml]: Removed in [d5842d1](https://github.com/ballerina-platform/persist-tools/pull/30/commits/d5842d197bb012124aad6e050948fe9edb602301)

File: persist-cli-tests/build.gradle [hunk 9]
```diff
@@ -26,26 +26,80 @@ description = 'Ballerina - persist-tools Tooling Test'
...
     checkstyle "com.puppycrawl.tools:checkstyle:${checkstylePluginVersion}"
 
-    testImplementation group: 'org.ballerinalang', name: 'ballerina-runtime', version: "${ballerinaLangVersion}"
-    testImplementation group: 'org.ballerinalang', name: 'ballerina-tools-api', version: "${ballerinaLangVersion}"
-    testImplementation (group: 'org.ballerinalang', name: 'ballerina-cli', version: "${ballerinaLangVersion}") {
-        exclude group: 'org.ballerinalang', module: 'maven-resolver'
+    implementation "org.ballerinalang:ballerina-lang:${ballerinaLangVersion}"
+    implementation "org.ballerinalang:ballerina-parser:${ballerinaLangVersion}"
+    implementation "org.ballerinalang:ballerina-cli:${ballerinaLangVersion}"
+    implementation "org.ballerinalang:formatter-core:${ballerinaLangVersion}"
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply persist-cli-tests/build.gradle]: Removed unnecessary dependencies in [420b9c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/420b9c2617c8de5f06973d312894d3f35cb4bc73). Some are needed for `persist-cli` module.

Formatter code is used to format the generated client bal file.

File: persist-cli/build.gradle (hunk 2 above)
Comment (MadhukaHarith92) [reply persist-cli/build.gradle:23]: Removed unnecessary dependencies in [420b9c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/420b9c2617c8de5f06973d312894d3f35cb4bc73)

**Comment by MadhukaHarith92** [comment]
> Lets clean up gitignore file too. I saw some files in target folder being committed

Those target folder were ignored from [gitignore](https://github.com/ballerina-platform/persist-tools/blob/ba096c3982f7ac14ba6d37daf954b2f14e50f23c/.gitignore#L27). But manually committed since they are needed for tests.

File: persist-cli-tests/build.gradle [hunk 10]
```diff
@@ -26,26 +26,80 @@ description = 'Ballerina - persist-tools Tooling Test'
...
+
+    implementation project(":persist-cli")
     testImplementation group: 'org.testng', name: 'testng', version: "${testngVersion}"
     implementation 'org.junit.jupiter:junit-jupiter:5.8.1'
     implementation (group: 'mysql', name: 'mysql-connector-java', version: "${mySqlDriverVersion}") {
         transitive = false
     }
 }
 
+def bDistribution = file("$project.buildDir/extracted-distribution/jballerina-tools-${ballerinaLangVersion}")
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply persist-cli-tests/build.gradle]: Updated in [cc080c2](https://github.com/ballerina-platform/persist-tools/pull/30/commits/cc080c2cd0a96d359e3d950202a76cb2637f7405)

File: .github/workflows/build-timestamped-master.yml [hunk 11]
```diff
@@ -34,15 +30,7 @@ jobs:
...
           packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
           packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
-          publishUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
-          publishPAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
         run: |
-          ./gradlew publish --scan --no-daemon
+          ./gradlew clean build publish --stacktrace --scan --console=plain
       - name: Generate CodeCov Report
-        uses: codecov/codecov-action@v2
-      - name: Upload Artifact
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply .github/workflows/build-timestamped-master.yml:43]: Added in [31f9e7d](https://github.com/ballerina-platform/persist-tools/pull/30/commits/31f9e7d86be841bd2e7eaa889e8a18937ec22661)

File: .github/workflows/build-timestamped-master.yml [hunk 12]
```diff
@@ -34,15 +30,12 @@ jobs:
...
           packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
           packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
-          publishUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
-          publishPAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
         run: |
-          ./gradlew publish --scan --no-daemon
+          ./gradlew clean build publish --stacktrace --scan --console=plain
       - name: Generate CodeCov Report
-        uses: codecov/codecov-action@v2
+        uses: codecov/codecov-action@v1
```
Comment (niveathika) [inline .github/workflows/build-timestamped-master.yml:36]:
```suggestion
        uses: codecov/codecov-action@v2
```

v1 is not compatible with jacoco plugin you added

File: .github/workflows/publish-release.yml [hunk 13]
```diff
@@ -2,66 +2,57 @@ name: Publish release
...
-          publishUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
-          publishPAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
         run: |
-          ./gradlew clean release -Prelease.useAutomaticVersion=true
+          ./gradlew release -Prelease.useAutomaticVersion=true
           ./gradlew -Pversion=${VERSION} publish -x test -PpublishToCentral=true
-      - name: GitHub Release and Release Sync PR
+      - name: Create Github release from the release tag
+        run: |
+          curl --request POST 'https://api.github.com/repos/ballerina-platform/persist-tools/releases' \
```
[the remark this answers is not in the document]
Comment (MadhukaHarith92) [reply .github/workflows/publish-release.yml]: Updated in [7b987db](https://github.com/ballerina-platform/persist-tools/pull/30/commits/7b987db7bb12b189fcc8041160285ad952a5e5de)

File: .github/workflows/publish-release.yml [hunk 14]
```diff
@@ -16,52 +14,36 @@ jobs:
...
           GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
           BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_ACCESS_TOKEN }}
           packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
           packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
-          publishUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
-          publishPAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
         run: |
-          ./gradlew clean release -Prelease.useAutomaticVersion=true
-          ./gradlew -Pversion=${VERSION} publish -x test -PpublishToCentral=true
+          ./gradlew release -Prelease.useAutomaticVersion=true
```
Comment (niveathika) [inline .github/workflows/publish-release.yml:42]: How are we publishing w/o publish PAT? `publishUser` `publishPAT`

Comment (MadhukaHarith92) [reply .github/workflows/publish-release.yml:44]: Added missing tokens in [44c62b3](https://github.com/ballerina-platform/persist-tools/pull/30/commits/44c62b3805a7c076543452da6b801e653da8381a)

Comment (niveathika) [reply .github/workflows/publish-release.yml:44]: Build workflow too publishes a timestamped jar. It needs the token

Comment (MadhukaHarith92) [reply .github/workflows/publish-release.yml:44]: Added in [92aee6e](https://github.com/ballerina-platform/persist-tools/pull/30/commits/92aee6e13125b698b4a58622a4e43b57523c1cf6)

**Review by niveathika — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

**Pull request merged**